File: Darts/src/code/screen/ai/AbstractAIConfigurationSubPanel.java
package code.screen.ai;

import javax.swing.JPanel;

import code.ai.AbstractDartsModel;
import code.ai.DartsModelNormalDistribution;

/**
 * 
 * @author alexb
 *
 */
public abstract class AbstractAIConfigurationSubPanel extends JPanel
{
	public abstract boolean valid();
	public abstract void populateModel(AbstractDartsModel model);
	public abstract void initialiseFromModel(AbstractDartsModel model);
	
	public void reset()
	{
		initialiseFromModel(new DartsModelNormalDistribution());
	}
}

File: Darts/src/main/kotlin/dartzee/screen/ai/AbstractAIConfigurationSubPanel.kt
package dartzee.screen.ai

import dartzee.ai.AbstractDartsModel
import dartzee.ai.DartsModelNormalDistribution
import javax.swing.JPanel

abstract class AbstractAIConfigurationSubPanel : JPanel()
{
    abstract fun valid(): Boolean
    abstract fun populateModel(model: AbstractDartsModel)
    abstract fun initialiseFromModel(model: AbstractDartsModel)

    fun reset()
    {
        initialiseFromModel(DartsModelNormalDistribution())
    }
}
